sql/src/test/java/io/crate/integrationtests/SQLTransportIntegrationTest.java

```diff
@@ -651,37 +653,74 @@ private boolean isJdbcEnabled() {
-     * based on the ratio provided, a random value of true or false is returned.
+     * If the Test class or method is annotated with {@link UseSemiJoins} then,
+     * based on the provided ratio, a random value of true or false is returned.
      * For more details on the ratio see {@link UseSemiJoins}
      * <p>
      * Method annotations have higher priority than class annotations.
      */
     private boolean isSemiJoinsEnabled() {
+        UseSemiJoins useSemiJoins= getTestAnnotation(UseSemiJoins.class);
+        if (useSemiJoins == null) {
+            return false;
+        }
+        return isFeatureEnabled(useSemiJoins.value());
+    }
+
+    /**
+     * If the Test class or method is annotated with {@link UseHashJoin} then,
+     * based on the provided ratio, a random value of true or false is returned.
+     * For more details on the ratio see {@link UseHashJoin}
+     * <p>
+     * Method annotations have higher priority than class annotations.
+     */
+    private boolean isHashJoinEnabled() {
+        UseHashJoin useHashJoin = getTestAnnotation(UseHashJoin.class);
+        if (useHashJoin == null) {
+            return false;
+        }
+        return isFeatureEnabled(useHashJoin.value());
+    }
+
+    /**
+     * Checks if the current test method or test class is annotated with the provided {@param annotationClass}
+     * @return the annotation if one is present or null otherwise
+     */
+    @Nullable
+    private <T extends Annotation> T getTestAnnotation(Class<T> annotationClass) {
         try {
             Class<?> clazz = this.getClass();
             Method method = clazz.getMethod(testName.getMethodName());
-            UseSemiJoins annotation = method.getAnnotation(UseSemiJoins.class);
+            T annotation = method.getAnnotation(annotationClass);
             if (annotation == null) {
-                annotation = clazz.getAnnotation(UseSemiJoins.class);
-                if (annotation == null) {
-                    return false;
-                }
+                annotation = clazz.getAnnotation(annotationClass);
             }
-            double ratio = annotation.value();
-            assert ratio >= 0.0 && ratio <= 1.0;
-            if (ratio == 0) {
-                return false;
-            }
-            if (ratio == 1) {
-                return true;
-            }
-            return RandomizedContext.current().getRandom().nextDouble() < ratio;
-        } catch (NoSuchMethodException ignored) {
+            return annotation;
+        } catch (NoSuchMethodException e) {
+            return null;
+        }
+    }
+
+    /**
+     * We sometimes randomize the use of features in tests.
+     * This method verifies the provided ratio and based on it and a random number
+     * indicates if a feature should be enabled or not.
+     *
+     * @param ratio a number between [0, 1] that indicates a "randomness factor"
+     *              (1 = mostly enabled, 0 = mostly disabled)
```

Comment:
1 = always enabled
0 = always disabled
Or you can rephrase to `towards 1` and `towards 0`